### ISSUE TITLE
[Possible feature request] Debounce processing a file for some amount of time

### ISSUE THREAD

**Issue by mgoo** [neoclide/coc.nvim#2749]
**Is your feature request related to a problem? Please describe.**

Far.vim opens all files that are going to be find-replaced. When find-replacing in multiple files, i can see the language server start up a process for each file. This hammers the CPU heaps when there are lots of files

**Describe the solution you'd like**

Possible solution is to not send the file to the language server if the focus changes within 0.5s.

Open to suggestions

**Describe alternatives you've considered**

Another possible solution is to wait until user input in the buffer before processing the file.

Using `:CocDisable` before doing the find and replace. Unfortunately the processes still start up. Maybe this should be the fix?

**Additional context**

All the processes for the language server ![image](https://user-images.githubusercontent.com/17482349/102824268-5f750680-4441-11eb-8b52-701509082f33.png)

**Comment by fannheyward** [comment]
> language server start up a process for each file.

Language server should not start a process for each file when edit multiple files.

From your screenshot, do you mean the `beam.smp` process is the language server?

**Comment by chemzqm** [comment]
>  i can see the language server start up a process for each file.

That should be bug, but I can't reproduce.

**Comment by mgoo** [comment]
beam.smp is a process of the language server yes. I used :CocInstall coc-elixir 

In a typescript project (CocInstall coc-tsserver I get lots (8+) eslint processes running in parallel. interestinly, only one tsserver process though. Perhaps all these processes are descendants of the main language server process

Aha, I have tracked down the problem. I forgot i have old plugin still enabled (thought it was disabled). This appears to be caused by the w0rp/ale plugin. Sorry about that
